app/components/nav.jsx

```diff
@@ -15,15 +16,53 @@ export class Nav extends Component {
     clearToken();
   }
 
+  constructor (props) {
+    super(props);
+    this.state = { loading: false };
+  }
+
+  componentWillMount () {
+    this.reset();
+  }
+
+  reset (props) {
+    const { retrieveUser, session } = props || this.props;
+
+    if (session) {
```

Comment:
wrapped this here to only do this stuff if `session`... not sure if this is the best place/way to do that.